nntrainer/tensor/manager.h

```diff
@@ -168,16 +168,30 @@ class Manager {
 
   /**
    * @brief Reset the manager state
+   * @note The tensors assigned to the layers are not reset. They will be
+   * automatically reset once the model is initialized again.
    */
   void reset() {
-    weights.clear();
-    max_grad_size = 0;
+    deallocateTensors(true);
+    weights_allocated = false;
+    tensors_allocated = false;
+
     total_weight_size = 0;
     total_grad_size = 0;
+    max_grad_size = 0;
+    max_derivative_size = 0;
     max_shared_inout = 0;
+
     weight_mmaped_memory.reset();
     grad_mmaped_memory.reset();
+
     in_outs.clear();
+    weights.clear();
+    is_act_type.clear();
+    is_flat_type.clear();
+
+    weights_initialized = false;
+    tensors_initialized = false;
```

Comment:
I think this tensors_initialized value setting should be inside of deallocateTensors.

Reply:
done.

Reply:
deallocateTensors does not de-initialize. So, IMO, tensors_initialized should not be modified inside `deallocateTensors`. Only allocation related variables must be edited.